api4/system.go

```diff
@@ -548,6 +553,108 @@ func getServerBusyExpires(c *Context, w http.ResponseWriter, r *http.Request) {
+	auditRec := c.MakeAuditRecord("upgradeToEnterprise", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+
+	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	if model.BuildEnterpriseReady == "true" {
+		c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.already-enterprise.app_error", nil, "", http.StatusTooManyRequests)
+		return
+	}
+
+	percentage, _ := c.App.Srv().UpgradeToE0Status()
+
+	if percentage > 0 {
+		c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.app_error", nil, "", http.StatusTooManyRequests)
+		return
+	}
+	if percentage == 100 {
+		c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.already-done.app_error", nil, "", http.StatusTooManyRequests)
+		return
+	}
+
+	if err := c.App.Srv().CanIUpgradeToE0(); err != nil {
+		var ipErr *upgrader.InvalidPermissions
+		var iaErr *upgrader.InvalidArch
+		switch {
+		case errors.As(err, &ipErr):
+			params := map[string]interface{}{
+				"MattermostUsername": ipErr.MattermostUsername,
+				"FileUsername":       ipErr.FileUsername,
+				"Path":               ipErr.Path,
+			}
+			if ipErr.ErrType == "invalid-user-and-permission" {
+				c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.invalid-user-and-permission.app_error", params, err.Error(), http.StatusForbidden)
+			} else if ipErr.ErrType == "invalid-user" {
+				c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.invalid-user.app_error", params, err.Error(), http.StatusForbidden)
+			} else if ipErr.ErrType == "invalid-permission" {
+				c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.invalid-permission.app_error", params, err.Error(), http.StatusForbidden)
+			}
+		case errors.As(err, &iaErr):
+			c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.system_not_supported.app_error", nil, err.Error(), http.StatusForbidden)
+		default:
+			c.Err = model.NewAppError("upgradeToEnterprise", "api.upgrade_to_enterprise.generic_error.app_error", nil, err.Error(), http.StatusForbidden)
+		}
+		return
+	}
+
+	c.App.Srv().Go(func() {
+		c.App.Srv().UpgradeToE0()
+	})
+
+	auditRec.Success()
+	w.WriteHeader(http.StatusAccepted)
+	ReturnStatusOK(w)
+}
+
+func upgradeToEnterpriseStatus(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	percentage, err := c.App.Srv().UpgradeToE0Status()
+	var s map[string]interface{}
+	if err != nil {
+		var isErr *upgrader.InvalidSignature
+		switch {
+		case errors.As(err, &isErr):
+			appErr := model.NewAppError("upgradeToEnterpriseStatus", "api.upgrade_to_enterprise_status.app_error", nil, err.Error(), http.StatusBadRequest)
```

Comment:
The red error text has the prefix "upgradeToEnterpriseStatus", which isn't appropriate formatting for an end user. The string shouldn't be included. See image.
<img width="1201" alt="upgradetoenterprisestatus text" src="https://user-images.githubusercontent.com/57730300/90690727-10b32800-e227-11ea-867b-ec983f6c76b7.png">

I'm not sure if this is being set here.

Reply:
I going to address all this errors from the frontend side instead of using the server side errors, so this PR can be merge without the need of this changes.

Reply:
Now should be fixed.